source/projects/private-voice-assistant/voice_remote_esp32.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: "Setting up an ESP32 to control Home Assistant"
+---
+
+This tutorial will guide you to turn an ESP32 (ATOM Echo by M5) into the
+world's most private voice assistant. Pick up the tiny device to talk to
+your smart home. Issue commands and get responses!
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* A controllable device in your Home Assistant network
+* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)
+* The password to your 2.4&nbsp;GHz Wi-Fi network
+* Chrome (or a Chromium-based browser like Edge) on desktop (not Android/iOS) 
+* ESP32 board with a built-in speaker: [M5Stack ATOM Echo Development Kit](https://shop.m5stack.com/products/atom-echo-smart-speaker-dev-kit?ref=NabuCasa)
+* USB-C cable to connect the ATOM Echo
+
+## Flashing the firmware onto the ATOM Echo
+
+1. Plug the USB-C cable into the M5Stack ATOM Echo Development Kit and connect it to your computer.
+1. Make sure this page is opened in a Chromium-based browser and select the **Connect** button below.
+
+   <script type="module" src="https://unpkg.com/esp-web-tools@9/dist/web/install-button.js?module"></script>
+   <esp-web-install-button manifest="https://firmware.esphome.io/voice-assistant/m5stack-atom-echo/manifest.json"></esp-web-install-button>
+
+   * In the pop-up window, select the USB serial port and select **Connect**.
+        * Select **Install Voice Assistant**, then **Install**.
+        * Follow the instructions provided by the installation wizard.
+        * When prompted, enter the credentials to your 2.4&nbsp;GHz Wi-Fi network.
+          * As the **Network name**, enter the SSID of your 2.4&nbsp;GHz Wi-Fi.
```

Comment:
There should be a drop down to pick your network, unless you select "Other". It can sometimes be that the device doesn't have enough time to scan for networks.